src/contract/event.rs

```diff
@@ -293,6 +285,196 @@ impl<T: Transport, E: Detokenize> Stream for EventStream<T, E> {
     }
 }
 
+/// Trait for parsing a transaction log into an some event data when the
+/// expected event type is not known.
+pub trait ParseLog: Sized {
+    /// Create a new instance by parsing raw log data.
+    fn parse_log(log: RawLog) -> Result<Self, ExecutionError>;
+}
+
+/// Raw log transaction data.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct RawLog {
+    /// The raw 32-byte topics.
+    pub topics: Vec<H256>,
+    /// The raw non-indexed data attached to an event.
+    pub data: Vec<u8>,
+}
+
+impl RawLog {
+    /// Decode raw log data into a tokenizable for a matching event ABI entry.
+    pub fn decode<D>(self, event: &AbiEvent) -> Result<D, ExecutionError>
+    where
+        D: Detokenize,
+    {
+        let event_log = event.parse_log(AbiRawLog {
+            topics: self.topics,
+            data: self.data,
+        })?;
+
+        let tokens = event_log
+            .params
+            .into_iter()
+            .map(|param| param.value)
+            .collect::<Vec<_>>()
+            .compat()
+            .ok_or(ExecutionError::UnsupportedToken)?;
+        let data = D::from_tokens(tokens)?;
+
+        Ok(data)
+    }
+}
+
+impl From<Log> for RawLog {
+    fn from(log: Log) -> Self {
+        RawLog {
+            topics: log.topics,
+            data: log.data.0,
+        }
+    }
+}
+
+impl ParseLog for RawLog {
+    fn parse_log(log: RawLog) -> Result<Self, ExecutionError> {
+        Ok(log)
+    }
+}
+
+/// A builder for creating a filtered stream for any contract event.
+#[must_use = "event builders do nothing unless you stream them"]
+pub struct AllEventsBuilder<T: Transport, E: ParseLog> {
```

Comment:
It's not possible to reuse the existing EventsBuilder here? Are allEvents and a specific event too different?

Reply:
There are a couple of issues:
- The type for the topic filter is different (because they need to account for the event signature in the other does not)
- `EventBuilder` needs an event entry from the ABI, `AllEventsBuilder` does not.
- They have different signatures for their `topicN` builder methods for specifying filters.